core/audits/byte-efficiency/legacy-javascript.js

```diff
@@ -229,23 +227,75 @@ class LegacyJavascript extends ByteEfficiencyAudit {
    * @return {Pattern[]}
    */
   static getTransformPatterns() {
+    /**
+     * @param {string} content
+     * @param {RegExp|string} pattern
+     * @return {number}
+     */
+    const count = (content, pattern) => {
+      // Split is slightly faster than match.
+      if (typeof pattern === 'string') {
+        return content.split(pattern).length;
```

Comment:
`'abc-abc-abc'.split('-')` gives us `['abc', 'abc', 'abc']` which has length 3 but there were only 2 matches for `-`

```suggestion
        return content.split(pattern).length - 1;
```